packages/hawtio/src/plugins/selectionNodeContext.ts

```diff
@@ -1,12 +1,27 @@
-import { createContext, useState } from 'react'
+import { createContext, useEffect, useState } from 'react'
 import { MBeanNode } from '@hawtiosrc/plugins/shared'
+import { selectionNodeDataService, MBeanAttributeData } from './selection-node-data-service'
 
 /**
  * Custom React hook for using JMX MBean tree.
  */
 export function usePluginNodeSelected() {
   const [selectedNode, setSelectedNode] = useState<MBeanNode | null>(null)
-  return { selectedNode, setSelectedNode }
+  const [selectedNodeAttributes, setSelectedNodeAttributes] = useState<MBeanAttributeData>({
+    nodeData: undefined,
+    children: {},
+  })
+  const [isReadingAttributes, setIsReadingAttributes] = useState<boolean>(false)
+
+  useEffect(() => {
+    ;(async () => {
+      setIsReadingAttributes(true)
+      setSelectedNodeAttributes(await selectionNodeDataService.getAttributesForNode(selectedNode))
+      setIsReadingAttributes(false)
+    })()
+  }, [selectedNode])
+
+  return { selectedNode, setSelectedNode, selectedNodeAttributes, isReadingAttributes }
 }
```

Comment:
Am not sure I am comfortable with attributes being collected here. The selectedNodeContext was at best a hack to get the same selection between the JMX and Camel trees. I don't think it advantageous to build on that hack.

If this is for the chart view only then it would be preferable to have <loading> tags then a useEffect in the Chart view to collect these attributes. This is the workflow that is used for many of the other views, eg. Debug, Trace, Routes, Endpoints.

Reply:
Ok, I removed the leftover code and left it as it was on main. That change was decided to not move forward and the feat doesn't need any of that. Thanks for spotting it :smile: